internal/aggregated-co/writer.go

```diff
@@ -0,0 +1,66 @@
+package aggregated_co
+
+import (
+	"context"
+	"reflect"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// From https://github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorstatus/writer.go
+
+// NewWriter returns a new instance of Writer.
+func NewWriter(client configv1client.ConfigV1Interface) *Writer {
+	return &Writer{
+		client: client,
+	}
+}
+
+// Writer encapsulates logic for cluster operator object API. It is used to
+// update ClusterOperator resource.
+type Writer struct {
+	client configv1client.ConfigV1Interface
+}
+
+// EnsureExists ensures that the cluster operator resource exists with a default
+// status that reflects expecting status.
```

Comment:
Looks like this function is also unused? (We're going to include the CO in our payload manifests, so I think it's safe to assume that it will exist)

Reply:
I think we've seen a couple of CO controller implementations (including OLM/marketplace) have logic for creating the resource during apierrors.IsNotFound(err) error checks. Not too sure whether that was a legacy issue, and that kind of logic is no longer needed, or it's something that the CVO team recommends as a best practice.

In any case, this kind of logic seems like it could be refactored to use c-r's CreateOrUpdate logic?

Reply:
I think the context here is that in OLM, this code was meant to safely run upstream, where no CO would be present. It's not being used by this PR, so I'm simply going to remove this function and keep things simple.

Reply:
Since the aggregate CO is part of the PO payload, we can rely on CVO to ensure it's always on-cluster.